src/lib/pickles/plonk_checks/scalars.ml

```diff
@@ -270,18 +654,977 @@ module Tick : S = struct
+                                               ) ) ) )
+                                 , fun () ->
+                                     field
+                                       "0x0000000000000000000000000000000000000000000000000000000000000000"
+                                 ) )
+                           * ( gamma
+                               * ( beta
+                                 + field
+                                     "0x0000000000000000000000000000000000000000000000000000000000000001"
+                                 )
+                             + cell (var (LookupTable, Curr))
+                             + (beta * cell (var (LookupTable, Next))) ) ) ) )
+                     ) )
+               , fun () ->
+                   field
+                     "0x0000000000000000000000000000000000000000000000000000000000000000"
+               ) ) )
+      ; ( LookupKindIndex ForeignFieldMul
+        , lazy
+            (if_feature
+               ( LookupTables
+               , (fun () ->
+                   alpha_pow 24
+                   * ( vanishes_on_last_4_rows
+                     * ( field
+                           "0x40000000000000000000000000000000224698FC094CF91B992D30ED00000000"
+                       * ( cell (var (LookupAggreg, Curr))
+                         * ( ( field
+                                 "0x40000000000000000000000000000000224698FC094CF91B992D30ED00000000"
+                               * if_feature
+                                   ( LookupPattern ForeignFieldMul
+                                   , (fun () ->
+                                       field
+                                         "0x0000000000000000000000000000000000000000000000000000000000000001"
+                                       )
+                                   , fun () ->
+                                       field
+                                         "0x0000000000000000000000000000000000000000000000000000000000000000"
+                                   )
+                               * ( if_feature
+                                     ( LookupsPerRow 1
+                                     , (fun () -> gamma)
+                                     , fun () ->
+                                         field
+                                           "0x0000000000000000000000000000000000000000000000000000000000000001"
+                                     )
+                                 * if_feature
+                                     ( LookupsPerRow 2
+                                     , (fun () -> gamma)
+                                     , fun () ->
+                                         field
+                                           "0x0000000000000000000000000000000000000000000000000000000000000001"
+                                     )
+                                 * if_feature
```

Comment:
Mmmmm... But in ForeignFieldMul, the maximum number of lookups per row is 2... Why do you include the case of 3 and 4 as well? Or if you do, doesn't it make sense to somehow cancel it out using the zero field element or similar? I guess a comment with the formula could be clarifying.